### PR TITLE
Add support for percentage-based thresholds

### PR DESCRIPTION
**Pull request by keatomue**
*Description of changes*

Adding support for `ThresholdExpression` on the AWS::CE::AnomalySubscription resource.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by kingsleyliao — APPROVED** [review]
lgtm

File: anomalysubscription/src/main/java/software/amazon/ce/anomalysubscription/RequestBuilder.java [hunk 1]
```diff
@@ -35,6 +37,7 @@ public static UpdateAnomalySubscriptionRequest buildUpdateAnomalySubscriptionReq
                 .subscriptionArn(model.getSubscriptionArn())
                 .subscriptionName(model.getSubscriptionName())
                 .threshold(model.getThreshold())
+                .thresholdExpression(model.getThresholdExpression() != null ? Utils.toExpressionFromJson(model.getThresholdExpression()) : null)
```
Comment (arodivya) [inline anomalysubscription/src/main/java/software/amazon/ce/anomalysubscription/RequestBuilder.java:40]: Could we add a comment to this method about acceptable inputs for Threshold and ThresholdExpression for updates

Comment (keatomue) [reply anomalysubscription/src/main/java/software/amazon/ce/anomalysubscription/RequestBuilder.java:40]: Yep, will add

File: anomalysubscription/src/test/java/software/amazon/ce/anomalysubscription/CreateHandlerTest.java [hunk 2]
```diff
@@ -68,6 +68,40 @@ public void handleRequest_Success() {
         assertThat(response.getErrorCode()).isNull();
     }
 
+    @Test
```
[the remark this answers is not in the document]
Comment (keatomue) [reply anomalysubscription/src/test/java/software/amazon/ce/anomalysubscription/CreateHandlerTest.java:71]: Sure, can add a similar unit test

**Review by arodivya — APPROVED** [review]

**Comment by ronkorving** [comment]
Did this never get support for `And` and `Or` expressions as described in some of the examples in the [CloudFormation docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ce-anomalysubscription.html#aws-resource-ce-anomalysubscription--examples)?

When I try to run this, I get an error when trying to create my changeset:

```yaml
    rCostAnomalySubscription:
      Condition: cIncludeCostAnomalyMonitor
      Type: AWS::CE::AnomalySubscription
      Properties:
        SubscriptionName: My team
        Frequency: DAILY
        ThresholdExpression:
          Fn::ToJsonString:
            And:
              - Dimensions:
                  Key: ANOMALY_TOTAL_IMPACT_PERCENTAGE
                  MatchOptions: [GREATER_THAN_OR_EQUAL]
                  Values: ["25"] # percent
              - Dimensions:
                  Key: ANOMALY_TOTAL_IMPACT_ABSOLUTE
                  MatchOptions: [GREATER_THAN_OR_EQUAL]
                  Values: ["25"] # USD
        MonitorArnList:
          - !Ref rCostAnomalyMonitor
        Subscribers:
          - Type: EMAIL
            Address: redacted@example.com
```

The error:
image

It seemed initially "clear" that I shouldn't use `Fn::ToJsonString`, but I was wrong. The moment I turned this into a JSON string and removed the `Fn::ToJsonString` function, the exact same error occurred.

**Comment by ronkorving** [comment]
I know CFN docs don't mention regions, but It would've saved me a lot of time if the docs had mentioned that. Thanks for the response here though, I do appreciate it.

**Comment by keatomue** [comment]
Hello! Been a while, but we just released to every region: https://aws.amazon.com/about-aws/whats-new/2023/12/cost-anomaly-detection-cloudformation-region-support/

Including, of course, `ap-northeast-1` like you were asking about.

Hopefully this helps!

**Pull request merged**